src/cpp/server/server_context.cc

```diff
@@ -283,12 +302,15 @@ void ServerContext::TryCancel() const {
 }
 
 bool ServerContext::IsCancelled() const {
-  if (has_notify_when_done_tag_) {
-    // when using async API, but the result is only valid
+  if (completion_tag_) {
+    // When using callback API, this result is always valid.
+    return completion_op_->CheckCancelledAsync();
+  } else if (completion_tag_ || has_notify_when_done_tag_) {
```

Comment:
} else if (has_notify_when_done_tag_) {

I haven't yet learned how to suggest code changes in github that author can immediately merge

Reply:
oops.